javascript/README.md

```diff
@@ -505,3 +505,36 @@ functionThatReturnsAPromise()
         // other code here!
     });
 ```
+
+## Module Definition
+
+When building a module that can be consumed by a number of different sources, we prefer
+to use [Universal Module Definition (UMD)](https://github.com/umdjs/umd), so that the
+module will be compatible with AMD, CommonJS, or plain script inclusion.
+For modules that are not shared across projects, UMD is not required.
```

Comment:
@jeremywiebe and @jansepar: Do you think that covers your discussion above? If so, let's get this merged.